Content.Server/Lathe/LatheSystem.cs

```diff
@@ -63,6 +68,41 @@ public override void Update(float frameTime)
             }
         }
 
+        private void OnLatherEjectMessage(EntityUid uid, LatheComponent lathe, LatheEjectMaterialMessage message)
+        {
+            int volume = message.WholeVolume;
+            if (!_prototypeManager.TryIndex<MaterialPrototype>(message.Material, out var material))
+            {
+                Logger.Error("Failed to index material prototype " + message.Material);
+                return;
+            }
```

Comment:
Server should validate client input in regards to volume, new loggers should also use a sawmill rather than the static logger, you should also just pass in the arg rather than using string concatenation e.g. `_sawmill.Error("Failed to index material prototype {0}", message.Material);`

Reply:
I changed a logger to sawmill and change string. About server check though, it's already checked by _materialStorage.TryChangeMaterialAmount this. If client will send crooked amount, this try will return false and the user will see an error.

Reply:
They can eject any amount they want, not just the button amounts.